dashboard/app/models/course.rb

```diff
@@ -257,10 +257,21 @@ def summarize(user = nil)
         script.summarize(include_stages).merge!(script.summarize_i18n(include_stages))
       end,
       teacher_resources: teacher_resources,
-      has_verified_resources: has_verified_resources?
+      has_verified_resources: has_verified_resources?,
+      versions: summarize_versions
     }
   end
 
+  # Returns an array of objects showing the name and version year for all courses
+  # sharing the assignment_family_name of this course, including this one.
+  def summarize_versions
+    Course.
+      where('name regexp ?', "^#{assignment_family_name}(-[0-9]{4})?$").
```

Comment:
I assume performance isn't a big deal here because this is a small set and I suspect we cache course summaries pretty heavily too.  Do you agree?

Reply:
We can cache course summaries by keying the cache on the experiments the user is in (or more specifically, the intersection of the set of experiments the user is in and the experiments that appear in the course's CourseScripts). I can imagine an A/B test that puts half (or perhaps all) of our users into a course experiment.

I'd also consider using the course cache to compute this version summary without hitting the DB, but this is a small enough query that it shouldn't matter much.

Reply:
Thanks Brad and Ram! Sounds like caching these requests is not a super high priority, and since we're on a super short timeline, I'll pass on any caching optimizations for now.